lib/broadcast/index.web.js

```diff
@@ -0,0 +1,35 @@
+const BROADCAST_ONYX = 'BROADCAST_ONYX';
+
+const subscriptions = [];
```

Comment:
There are many cases when we're adding subscriptions, but we never actually remove them. This means as you use the app this array will grow without bounds.

To address this, we should have `subscribe` return an `unsubscribe` cleanup function that removes the callback from the array:

```js
/**
 * Subscribes to the broadcast channel. Every time a new message
 * is received, the callback is called.
 *
 * @param {Function} callback
 * @returns {Function} unsubscribe
 */
function subscribe(callback) {
    subscriptions.push(callback);
    channel.onmessage = (message) => {
        subscriptions.forEach(c => c(message));
    };
    return () => {
        const index = subscriptions.indexOf(callback);
        subscriptions.splice(index, 1);
    };
}
```

Then in all places where this was used, keep a reference to all the cleanup functions and call them in a `beforeunload` handler.